src/main/java/org/jabref/logic/importer/fetcher/DoiFetcher.java

```diff
@@ -123,4 +133,18 @@ private void doPostCleanup(BibEntry entry) {
 
         return agency;
     }
+
+    private void setPageCountToArticleId(BibEntry entry, String doiAsString) {
+        String articleId = doiAsString.substring(doiAsString.lastIndexOf('.') + 1);
+        entry.setField(StandardField.PAGES, articleId);
+    }
+
+    private boolean isAPSJournal(BibEntry entry, String doiAsString) {
+        if (!entry.getType().equals(StandardEntryType.Article)) {
+            return false;
+        }
+        Pattern apsJournalSuffixPattern = Pattern.compile("([\\w]+\\.)([\\w]+\\.)([\\w]+)");
```

Comment:
I am thinking if it would make sense to check for the strig "phys" as well?

It's better to etract the Pattern to a static string, otherwise you won't get the advantages

Reply:
Can we be sure that every Physical Review journal contains that string? For example, this aps doi doesn't contain the string "phys". `https://doi.org/10.1103/PRXQuantum.1.010001`
Here is a list of all journals
https://journals.aps.org/browse

Reply:
I think the best is to check for the string/number 10.1103/  It seems all DOIs from APS are prefixed with that.
Otherwise the regex is too broad and would capture a lot of non related things and would create invalid data

Reply:
In that case, a regex isn't really needed because in the string 10.1103, `1103` denotes the organization (APS in this case). That's really all that has to be checked if I am not mistaken.

Reply:
I still would use the regex as you can test that the doi is of the right format, so that you really know that the last number is the page number. And it's also a bit easier to understand than the manual parsing using isDigit

Reply:
So we
1. use the organization id to check if the entry is an aps journal
2. use the regex to check if the doi is of the right format (https://doi.org/10.1103/[journal].[volume].[articleID])
3. set the page field if 1 and 2 are true